b+tree.ts

```diff
@@ -488,6 +488,213 @@ export default class BTree<K=any, V=any> implements ISortedMapF<K,V>, ISortedMap
     return {nodequeue, nodeindex, leaf:nextnode};
   }
 
+  /**
+   * Computes the delta operations (adds, deletes, and modifications) required to go from this tree to the supplied destination.
+   * For efficiency, the delta is returned via invocations of supplied handlers.
+   * The computation is optimized for the case in which the two trees have large amounts of shared data (obtained by calling the
+   * `clone` or `with` APIs) and will avoid any iteration of shared state.
+   * @param destination The tree to compute a delta to.
+   * @param processAdd Callback invoked for all adds in the delta.
+   * @param processDelete Callback invoked for all deletions in the delta.
+   * @param processModify Callback invoked for all modifications in the delta.
+   */
+  delta(destination: BTree<K, V>, processAdd: (k: K, v: V) => void, processDelete: (k: K, v: V) => void, processModify: (k: K, vOld: V, vNew: V) => void): void {
```

Comment:
One usecase for this function (which we actually need) is an efficient equality implementation.

As it currently works, there is no way to early exit once a difference is found.

I think the difference handlers should have a return type which permits signaling the whole algorithm to early exit.

One way to do this which also covers some other use cases is to have a return value thats generic. Ex: Foo | undefined, and if something other than undefined is returned, end the algorithm and return the Foo from the top level function.

Other approach would be to have a special STOP sentinel value you could return.